### PR TITLE
【Hackathon 6th No.16】为 Paddle 新增 LPPool1D / LPPool2D API

### PR DESCRIPTION
**Pull request by Asthestarsfalll**
- https://github.com/PaddlePaddle/Paddle/pull/63544

### CODE REVIEW

File: docs/api/paddle/nn/LPPool1D_cn.rst [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    Output(N_i, C_i, l) = sum(Input[N_i, C_i, stride \times l:stride \times l+k]^{norm\_type})^{1/norm\_type}
+
+参数
+:::::::::
+    - **norm_type** (int|float)：幂平均池化的指数，不可以为 0。
+    - **kernel_size** (int|list|tuple) - 池化核的尺寸大小。如果 kernel_size 为 list 或 tuple 类型，其必须包含一个整数，最终池化核的大小为该数值。
+    - **stride** (int|list|tuple，可选) - 池化操作步长。如果 stride 为 list 或 tuple 类型，其必须包含一个整数，最终池化操作的步长为该数值。默认值为 None，这时会使用 kernel_size 作为 stride。
+    - **padding** (str|int|list|tuple，可选) - 池化补零的方式。如果 padding 是一个字符串，则必须为 `SAME` 或者 `VALID`。如果是 turple 或者 list 类型，则应是 `[pad_left, pad_right]` 形式。如果 padding 是一个非 0 值，那么表示会在输入的两端都 padding 上同样长度的 0。默认值为 0。
+    - **ceil_mode** (bool，可选) - 是否用 ceil 函数计算输出的 height 和 width，如果设置为 False，则使用 floor 函数来计算，默认为 False。
+    - **data_format** (str)：输入和输出的数据格式，可以是"NCL"和"NLC"。N 是批尺寸，C 是通道数，L 是特征长度。默认值："NCL"
```
Comment (sunzhongkai588) [inline docs/api/paddle/nn/LPPool1D_cn.rst]:
```suggestion
    - **data_format** (str，可选)：输入和输出的数据格式，可以是"NCL"和"NLC"。N 是批尺寸，C 是通道数，L 是特征长度。默认值："NCL"
```

File: docs/api/paddle/nn/LPPool1D_cn.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+.. math::
+
+    Output(N_i, C_i, l) = sum(Input[N_i, C_i, stride \times l:stride \times l+k]^{norm\_type})^{1/norm\_type}
+
+参数
+:::::::::
+    - **norm_type** (int|float)：幂平均池化的指数，不可以为 0。
+    - **kernel_size** (int|list|tuple) - 池化核的尺寸大小。如果 kernel_size 为 list 或 tuple 类型，其必须包含一个整数，最终池化核的大小为该数值。
+    - **stride** (int|list|tuple，可选) - 池化操作步长。如果 stride 为 list 或 tuple 类型，其必须包含一个整数，最终池化操作的步长为该数值。默认值为 None，这时会使用 kernel_size 作为 stride。
+    - **padding** (str|int|list|tuple，可选) - 池化补零的方式。如果 padding 是一个字符串，则必须为 `SAME` 或者 `VALID`。如果是 turple 或者 list 类型，则应是 `[pad_left, pad_right]` 形式。如果 padding 是一个非 0 值，那么表示会在输入的两端都 padding 上同样长度的 0。默认值为 0。
```
Comment (sunzhongkai588) [inline docs/api/paddle/nn/LPPool1D_cn.rst]: padding 在英文 docstring 里写了五种情况，这里似乎对不上？尽量中英文同步

File: docs/api/paddle/nn/LPPool2D_cn.rst [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    out(N_i, C_j, h, w)  = (\sum_{m=0}^{ksize[0]-1} \sum_{n=0}^{ksize[1]-1}
+                               input(N_i, C_j, stride[0] \times h + m, stride[1] \times w + n)^{norm\_type})^{1 / norm\_type}
+
+
+参数
+:::::::::
+    - **norm_type** (int|float)：幂平均池化的指数，不可以为 0。
+    - **kernel_size** (int|list|tuple)：池化核大小。如果它是一个元组或列表，它必须包含两个整数值，(pool_size_Height, pool_size_Width)。若为一个整数，则它的平方值将作为池化核大小，比如若 pool_size=2，则池化核大小为 2x2。
+    - **stride** (int|list|tuple，可选)：池化层的步长。如果它是一个元组或列表，它将包含两个整数，(pool_stride_Height, pool_stride_Width)。若为一个整数，则表示 H 和 W 维度上 stride 均为该值。默认值为 None，这时会使用 kernel_size 作为 stride。
+    - **padding** (str|int|list|tuple，可选) 池化填充。如果它是一个字符串，可以是"VALID"或者"SAME"，表示填充算法，计算细节可参考上述 pool_padding = "SAME"或 pool_padding = "VALID" 时的计算公式。如果它是一个元组或列表，它可以有 3 种格式：(1)包含 2 个整数值：[pad_height, pad_width]；(2)包含 4 个整数值：[pad_height_top, pad_height_bottom, pad_width_left, pad_width_right]；(3)包含 4 个二元组：当 data_format 为"NCHW"时为 [[0,0], [0,0], [pad_height_top, pad_height_bottom], [pad_width_left, pad_width_right]]，当 data_format 为"NHWC"时为[[0,0], [pad_height_top, pad_height_bottom], [pad_width_left, pad_width_right], [0,0]]。若为一个整数，则表示 H 和 W 维度上均为该值。默认值：0。
```
Comment (sunzhongkai588) [inline docs/api/paddle/nn/LPPool2D_cn.rst]: `计算细节可参考上述 pool_padding = "SAME"或 pool_padding = "VALID" 时的计算公式。` 上述的 pool_padding 是啥

File: docs/api/paddle/nn/LPPool2D_cn.rst [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+输入：
+
+    X 形状：:math:`\left ( N,C,H_{in},W_{in} \right )`
+
+属性：
+
+    kernel_size: :math:`ksize`
+    stride: :math:`stride`
+    norm_type: :math:`norm\_type`
```
Comment (sunzhongkai588) [inline docs/api/paddle/nn/LPPool2D_cn.rst]:
```suggestion
    - kernel_size: :math:`ksize`
    - stride: :math:`stride`
    - norm_type: :math:`norm\_type`
```

File: docs/api/paddle/nn/functional/lp_pool1d_cn.rst [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+   详细请参考对应的 `Class` 请参考：:ref:`cn_api_paddle_nn_LPPool1D` 。
+
+
+参数
+:::::::::
+    - **x** (Tensor)：当前算子的输入，其是一个形状为 `[N, C, L]` 的 3-D Tensor。其中 `N` 是 batch size, `C` 是通道数，`L` 是输入特征的长度。其数据类型为 float16、float32 和 float64。
+    - **norm_type** (int|float)：幂平均池化的指数，不可以为 0。
+    - **kernel_size** (int|list|tuple)：池化核的尺寸大小。如果 kernel_size 为 list 或 tuple 类型，其必须包含一个整数。
+    - **stride** (int|list|tuple)：池化操作步长。如果 stride 为 list 或 tuple 类型，其必须包含一个整数。
+    - **padding** (string|int|list|tuple)：池化补零的方式。如果 padding 是一个字符串，则必须为 `SAME` 或者 `VALID`。如果是 turple 或者 list 类型，则应是 `[pad_left, pad_right]` 形式。如果 padding 是一个非 0 值，那么表示会在输入的两端都 padding 上同样长度的 0。
```
Comment (sunzhongkai588) [inline docs/api/paddle/nn/functional/lp_pool1d_cn.rst:21]: `kernel_size` `stride` `padding` 这几个参数描述和英文都对不上，务必仔细检查一下 （包括其他参数描述） https://github.com/PaddlePaddle/Paddle/pull/63544/files#diff-6081ec42b042ea78a3fe77f89a03eb36958fc338aa77ae5ca2d38594d58ab1c2

File: docs/api/paddle/nn/functional/lp_pool2d_cn.rst [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+输入：
+
+    X 形状：:math:`\left ( N,C,H_{in},W_{in} \right )`
+
+属性：
+
+    kernel_size: :math:`ksize`
+    stride: :math:`stride`
+    norm_type: :math:`norm\_type`
```
Comment (sunzhongkai588) [inline docs/api/paddle/nn/functional/lp_pool2d_cn.rst]:
```suggestion
    - kernel_size: :math:`ksize`
    - stride: :math:`stride`
    - norm_type: :math:`norm\_type`
```

File: docs/api/paddle/nn/functional/lp_pool2d_cn.rst [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+输出：
+
+    Out 形状：:math:`\left ( N,C,H_{out},W_{out} \right )`
+
+.. math::
+    out(N_i, C_j, h, w)  = (\sum_{m=0}^{ksize[0]-1} \sum_{n=0}^{ksize[1]-1}
+                               input(N_i, C_j, stride[0] \times h + m, stride[1] \times w + n)^{norm\_type})^{1 / norm\_type}
+
+
+参数
```
Comment (sunzhongkai588) [inline docs/api/paddle/nn/functional/lp_pool2d_cn.rst:31]: 这边的参数的一些描述也和英文对齐吧，比如 `pool_size_Height`..

**Review by sunzhongkai588 — COMMENTED** [review]
在 paddle.nn.Overview 加一下描述

File: docs/api/paddle/nn/functional/lp_pool2d_cn.rst (hunk 7 above)
Comment (Asthestarsfalll) [reply docs/api/paddle/nn/functional/lp_pool2d_cn.rst:31]: 这里应该和上面的属性保持一致吧

File: docs/api/paddle/nn/LPPool2D_cn.rst (hunk 3 above)
Comment (Asthestarsfalll) [reply docs/api/paddle/nn/LPPool2D_cn.rst]: 文档直接修改自AvgPool，，，然而找不到这个东西

File: docs/api/paddle/nn/functional/lp_pool2d_cn.rst (hunk 7 above)
Comment (sunzhongkai588) [reply docs/api/paddle/nn/functional/lp_pool2d_cn.rst:31]: > 这里应该和上面的属性保持一致吧

属性里写的好像也是 `kernal_size` ？

File: docs/api/paddle/nn/Overview_cn.rst
Comment (sunzhongkai588) [inline docs/api/paddle/nn/Overview_cn.rst:1]: 也加上 `paddle.nn.functional.lp_pool1d` 和 `paddle.nn.functional.lp_pool2d`

File: docs/api/paddle/nn/functional/lp_pool2d_cn.rst (hunk 7 above)
Comment (Asthestarsfalll) [reply docs/api/paddle/nn/functional/lp_pool2d_cn.rst:31]: - kernel_size: :math:`ksize`

kernel_size应该是参数名，后面的是属性吧？

**Review by sunzhongkai588 — APPROVED** [review]
LGTM，修复一下CI错误
